synadm/api.py

```diff
@@ -556,6 +556,8 @@ def user_modify(self, user_id, password, display_name, threepid,
             data.update({"deactivated": True})
         if deactivation == "activate":
             data.update({"deactivated": False})
+        if user_type:
+            data.update({"user_type": None if 'null' else user_type})
```

Comment:
```suggestion
            data.update({"user_type": None if user_type == 'null' else
                         user_type})
```

`'null'` always evaluates to true, so user type will always be reset regardless of passed value.